src/coreclr/vm/debugdebugger.cpp

```diff
@@ -1078,6 +1082,70 @@ StackWalkAction DebugStackTrace::GetStackFramesCallback(CrawlFrame* pCf, VOID* d
 }
 #endif // !DACCESS_COMPILE
 
+PTR_VOID DebugStackTrace::GetExactGenericArgsToken(PTR_MethodDesc pFunc, PREGDISPLAY pRD, EECodeInfo* pCodeInfo, PTR_Frame pFrame)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_COOPERATIVE;
+    }
+    CONTRACTL_END;
+
+    _ASSERTE(pRD != NULL && pCodeInfo != NULL);
+
+    if (!pFunc || !pFunc->IsSharedByGenericInstantiations())
+        return NULL;
+
+    if (pFunc->AcquiresInstMethodTableFromThis())
+    {
+        return NULL;
+        
+        // TODO: how to avoid the possible GC hole here?
```

Comment:
Not sure how to fix it currently. The `pCodeInfo->GetCodeManager()->GetInstance(pRD, pCodeInfo)` can produce a corrupted `OBJECTREF` here.

Reply:
The generic context is used for very limited purpose today. It is quite possible that the generic context reporting in the JIT is only accurate for that limited purpose today.

Reply:
Is there any way to try getting the `this` but skip verifying whether it's valid or not?